lib/translator/read.cc

```diff
@@ -109,26 +103,24 @@ StatusCode Read(uint8_t rd_protect, bool fua, uint64_t lba,
     DebugLog("Transfer length exceeds limit of 16 bits");
     return StatusCode::kInvalidInput;
   }
-  transfer_length &= 0xffff;
+  transfer_length &= 0xffff;  // truncate to 16 bits
 
   StatusCode status =
-      LegacyRead(lba, nvme_cmd, allocation, nsid,
+      LegacyRead(nvme_cmd, allocation, nsid,
                  GetTransferLengthPages(transfer_length, page_size, lba_size));
-
   if (status != StatusCode::kSuccess) {
     return status;
   }
 
   uint8_t prinfo;  // Protection Information field 4 bits;
   status = BuildPrinfo(rd_protect, prinfo);
-
   if (status != StatusCode::kSuccess) {
     return status;
   }
 
-  nvme_cmd.cdw[2] = BuildCdw12(transfer_length, prinfo, fua);
+  nvme_cmd.cdw[2] = __bswap_32(htonl(BuildCdw12(transfer_length, prinfo, fua)));
```

Comment:
If you're going to do a double byteswap to guarantee that the variable is little endian, please define a function to do so.

Reply:
good idea, defined helper functions here https://github.com/googleinterns/scsi2nvme/pull/157

Reply:
Looks good!